libraries/discriminator/src/discriminator.rs

```diff
@@ -79,3 +76,26 @@ impl From<ArrayDiscriminator> for u64 {
         u64::from_le_bytes(from.0)
     }
 }
+
+#[cfg(feature = "borsh")]
+impl borsh::de::BorshDeserialize for ArrayDiscriminator
+where [u8; 8]: borsh::de::BorshDeserialize
+{
+    fn deserialize(
+        buf: &mut &[u8],
+    ) -> ::core::result::Result<Self, borsh::maybestd::io::Error> {
+        borsh::BorshDeserialize::deserialize(buf)
+    }
+}
+
+#[cfg(feature = "borsh")]
+impl borsh::ser::BorshSerialize for ArrayDiscriminator
+where [u8; 8]: borsh::ser::BorshSerialize
+{
+    fn serialize<W: borsh::maybestd::io::Write>(
+        &self,
+        writer: &mut W,
+    ) -> ::core::result::Result<(), borsh::maybestd::io::Error> {
+        borsh::BorshSerialize::serialize(&self.0, writer)
+    }
+}
```

Comment:
It'll be easier to do `#[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]` rather than deriving these by hand

https://stackoverflow.com/questions/42046327/conditionally-derive-based-on-feature-flag

Reply:
oh nice! Didn't know you could do this